mnms runs with mbatch

iabrilcabezas opened the pull request:
need to run noise_gen and noise_sim.py scripts. I am using mbatch.

[mbatch](https://github.com/simonsobs/mbatch/tree/main) requires --output-dir argument (where .sh and output messages of run are store). also need to transform the args.qid argument because it is read in as a string. I believe these changes should not alter current pipelines that run mnms/scripts/

iabrilcabezas committed:
change how args.qid is read to be able to run scripts using mbatch

iabrilcabezas committed:
Merge branch 'main' of github.com:simonsobs/mnms

zatkins2 reviewed (changes requested):
Hi Irene, could you add your `--output-dir` argument to `noise_all.py` as well?

Can you elaborate more on the change to `args.qid`? The way it was before I think does the same thing: the options supplied to `parser.add_argument` causes `args.qid` to be a list of strings

iabrilcabezas commented:
it's related to this problem:
![Captura de pantalla 2024-10-07 a las 9 28 35](https://github.com/user-attachments/assets/3f4caef3-0cc6-487f-bda1-f76965a7ab25)

You can reproduce this error with a small script `test_mbatch.py` like this one:

```python

from mnms import noise_models as nm
import argparse

parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)

parser.add_argument("--output-dir", dest="output_dir", type=str, default=None)

parser.add_argument('--config-name', dest='config_name', type=str, required=True,
                    help='Name of model config file from which to load parameters')

parser.add_argument('--noise-model-name', dest='noise_model_name', type=str, required=True,
                    help='Name of model within config file from which to load parameters')

parser.add_argument('--qid', dest='qid', nargs='+', type=str, required=True,
                    help='list of soapack array "qids"')

args = parser.parse_args()
print(args.qid)
# args.qid = [item for sublist in args.qid for item in sublist.split()]

model = nm.BaseNoiseModel.from_config(args.config_name, args.noise_model_name,
                                      *args.qid)
print('success')
```

To run it, we use a test mbatch config file looks like this
```yaml

root_dir: /home/ia404/gitreps/mnms/scripts/

stages:

  stage_test:
    exec: python
    options:
      config-name: 'act_dr6v4_day'
      noise-model-name: 'tile_cmbmask_daywide'
      qid:
       - 'pa5a_dw'
       - 'pa6a_dw'

    script: scripts/test_mbatch.py
```

This generates 
`python scripts/test_mbatch.py --config-name=act_dr6v4_day --noise-model-name=tile_cmbmask_daywide --qid="['pa5a_dw', 'pa6a_dw']" --output-dir /home/ia404/gitreps/mnms/scripts/test/stage_test`, which doesn't work because qid ends up being `["['pa5a_dw', 'pa6a_dw']"]`, pa5a and pa6a all together.

Another alternative: 
```yaml

root_dir: /home/ia404/gitreps/mnms/scripts/

stages:

  stage_test:
    exec: python
    options:
      config-name: 'act_dr6v4_day'
      noise-model-name: 'tile_cmbmask_daywide'
      qid: 'pa5a_dw pa6a_dw'

    script: scripts/test_mbatch.py
```

But in this case we have
`python scripts/test_mbatch.py --config-name=act_dr6v4_day --noise-model-name=tile_cmbmask_daywide --qid='pa5a_dw pa6a_dw' --output-dir /home/ia404/gitreps/mnms/scripts/test/stage_test`

which also doesn't work, `['pa5a_dw pa6a_dw']` not separated again.

Adding 
`args.qid = [item for sublist in args.qid for item in sublist.split()]` is the best solution I could find for it to be able to run with mbatch, while also being compatible with older scripts.

iabrilcabezas committed:
output-dir on noise_all.py

zatkins2 commented:
Ok, this is very clear. I still wonder if there is a cleaner solution. Can you test what happens in `mbatch` if you supply `qid` more than once in a config file? 

```
qid: 'pa5a_dw'
qid: 'pa6a_dw'

zatkins2 commented:
Please also target `develop` :)

iabrilcabezas committed:
patch fix on noise_all too

iabrilcabezas commented:
It's now targeting develop, and I opened an issue on mbatch, hopefully good to merge now --

zatkins2 commented:
As discussed offline, we will hold off on merging until https://github.com/simonsobs/mbatch/issues/25 is fixed on the `mbatch` side.

zatkins2 committed:
bump version, README updates: no install-time circular dependency with sofind

zatkins2 committed:
intermediate changes

zatkins2 committed:
noise_models: add cal to cache, remove mul from DataManager, add get_cal, add calibrate and calibrations_subproduct kwargs to get_sim and change seed kwarg default to 'auto'. utils: remove get_mul_fact. io: add calibrations_subproduct and change meaning of calibrated

zatkins2 committed:
bump version

zatkins2 committed:
cals can broadcast correctly if alm=True in get_sim

zatkins2 committed:
Merge pull request #13 from simonsobs/post_calibrate

Support "post sim calibration" in noise models

iabrilcabezas committed:
Merge pull request #14 from simonsobs/main

update with main

iabrilcabezas committed:
fix to "okmap becomes complex164 if left unattended"

iabrilcabezas committed:
Merge pull request #16 from simonsobs/dtype-mismatch

fix to "okmap becomes complex164 if left unattended"

zatkins2 committed:
use get_super_qids_from_qids_and_subproduct_kwargs from s_utils in DataManager instead of doing it ourselves

zatkins2 committed:
inpaint radius and thumb_width passed in configs

zatkins2 committed:
rand_alm default chunks to preserve random draws for changing preshape. also rand_alm lim, lim0 default set to 0

zatkins2 committed:
Merge branch 'catalog_args_in_config'

zatkins2 committed:
bump version

iabrilcabezas committed:
Merge branch 'main' into mbatch
